crates/database/interface/src/async_db.rs

```diff
@@ -13,7 +13,7 @@ use crate::{Database, DatabaseRef};
 /// Use [WrapDatabaseAsync] to provide [Database] implementation for a type that only implements this trait.
 pub trait DatabaseAsync {
     /// The database error type.
-    type Error: Send;
+    type Error: core::error::Error + Send;
```

Comment:
I would consider adding a `'static` bound here, as described in https://github.com/bluealloy/revm/pull/1840/files#r1819000225